optuna/trial.py

```diff
@@ -357,21 +411,32 @@ def report(self, value, step=None):
 
         Example:
 
-            Report intermediate scores of `SGDClassifier <https://scikit-learn.org/stable/modules/
-            generated/sklearn.linear_model.SGDClassifier.html>`_ training
+            .. testcode::
+
+                import optuna
+                import numpy as np
+                from sklearn.linear_model import SGDClassifier
+                from sklearn.model_selection import train_test_split
```

Comment:
This example has more than 20 lines, and `trial.report` may not be emphasized enough. 
It is controversial, but I think we can use the `testsetup` directive to hide unimportant lines such as the module imports and the dataset preparation.


Here's an example code and rendering result.

```python
            .. testsetup::

                import optuna
                import numpy as np
                from sklearn.linear_model import SGDClassifier
                from sklearn.model_selection import train_test_split

                np.random.seed(seed=0)
                X = np.random.randn(50).reshape(-1, 1)
                y = np.random.randint(0, 2, 50)
                X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

            .. testcode::

                def objective(trial):
```

![image](https://user-images.githubusercontent.com/3255979/73531665-485eac00-445e-11ea-9540-01a1569e9fd0.png)

Reply:
I think your solution is better than mine. Output document is more understandable.
BTW, should I apply this to other examples too?

Reply:
Thanks!

> BTW, should I apply this to other examples too?

I don't think so. The `trial.report` example is essentially long because it requires a dataset and a machine learning model to be viable and realistic. On the other hand, the suggest APIs such as `trial.suggest_uniform` and `trial.suggest_int` do not require such preparation. So, I think all examples for them can be simplified like the `trial.suggest_uniform` example and they will be short enough.